nixos/tests/kexec.nix

```diff
@@ -18,28 +16,32 @@ import ./make-test-python.nix ({ pkgs, lib, ... }: {
 
     node2 = { modulesPath, ... }: {
       virtualisation.vlans = [ ];
+      environment.systemPackages = [ pkgs.hello ];
       imports = [
         "${modulesPath}/installer/kexec/kexec-boot.nix"
+        "${modulesPath}/profiles/minimal.nix
       ];
     };
   };
 
   testScript = { nodes, ... }: ''
+    # Test whether reboot via kexec works.
     node1.wait_for_unit("multi-user.target")
     node1.succeed('kexec --load /run/current-system/kernel --initrd /run/current-system/initrd --command-line "$(</proc/cmdline)"')
     node1.execute("systemctl kexec >&2 &", check_return=False)
     node1.connected = False
     node1.connect()
     node1.wait_for_unit("multi-user.target")
 
-    # Check the machine with kexec-boot.nix profile boots up
+    # Check if the machine with kexec-boot.nix profile boots up
     node2.wait_for_unit("multi-user.target")
     node2.shutdown()
 
     # Kexec node1 to the toplevel of node2 via the kexec-boot script
     node1.succeed('touch /run/foo')
     node1.execute('${nodes.node2.config.system.build.kexecBoot}/kexec-boot', check_return=False)
     node1.succeed('! test -e /run/foo')
+    node1.succeed('hello')
```

Comment:
`hello` could end up in a configuration by accident. Could you check `node1.fail('hello')` before kexec?

You could also check `node1.succeeds('diff -U1 --label actual <(hostname) --label expected <(echo node2)')`

Reply:
is that easier to read than `[ "$(hostname)" = "node2" ]` ?

Reply:
Nope. Don't know why I made it complicated. Go ahead :)

Reply:
Ah I forgot that the python runner will print the command if it fails. That's why.
Yours is better.